### PR TITLE
Scalafmt + Warnings

### PR DESCRIPTION
**Pull request by benkio**
I noticed the repository doesn't have the `scalafmt` plugin. 

I set it up and run it + I run `printWarnings` and fixed a little deprecation.
The `scalafmt` configuration is copied from `cats` repo

### CODE REVIEW

**Comment by johnspade** [comment]
Thank you!

Please exclude telegramium-core, it's generated code. We already format it with [this scalafmt.conf](https://github.com/apimorphism/iozhik/blob/152db9a5e22deb182b38760e73afbd80ff190893/iozhtest/src/main/resources/.scalafmt.conf), let's use the same config for Telegramium with `maxColumn = 120`.

**Comment by benkio** [comment]
Done

File: .scalafmt.conf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version = "3.0.0-RC4"
+align.preset = true
+maxColumn = 120
+newlines.alwaysBeforeTopLevelStatements = true
+# verticalMultiline.newlineAfterOpenParen = true
+runner.debug = true
```
Comment (johnspade) [inline .scalafmt.conf:6]: It's not needed

File: .scalafmt.conf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+version = "3.0.0-RC4"
```
Comment (johnspade) [inline .scalafmt.conf:1]: Looks like a lot of defaults have changed, it's better to use `version=1.5.1`

**Review by johnspade — CHANGES_REQUESTED** [review]

File: .scalafmt.conf (hunk 2 above)
Comment (benkio) [reply .scalafmt.conf:1]: Are they objectively bad or it's just a personal taste?
I'll do the change

Comment (johnspade) [reply .scalafmt.conf:1]: They're fine, I just want to unify the formatting with telegramium-core.

Comment (benkio) [reply .scalafmt.conf:1]: That makes sense, then can't we just bring the `telegramium-core` up to `scalafmt` 3.0,0? or there's something that take us down?

Comment (johnspade) [reply .scalafmt.conf:1]: Can you make a PR to https://github.com/apimorphism/iozhik then? I like this config, what do you think?

```
version="3.0.0-RC5"
align.preset=more
maxColumn = 120
align.openParenCallSite = true
align.openParenDefnSite = true
continuationIndent.defnSite = 2
assumeStandardLibraryStripMargin = true
newlines.afterCurlyLambda = preserve
danglingParentheses.preset = true
importSelectors = singleLine
```

Comment (benkio) [reply .scalafmt.conf]: I did the change here but the master of `iozhik` is not compiling.
This is the error I got by `sbt compile`
```
/home/benkio/workspace/iozhik/javatest/target/scala-2.12/classes ...
[error] /home/benkio/workspace/iozhik/iozhtest/src/main/scala/iozhik/vk/Poll.scala:25:18: Poll is already defined as case class Poll
[error] final case class Poll(id: Int,
[error]                  ^
[error] one error found
```

Comment (johnspade) [reply .scalafmt.conf]: Yeah, just replace the config

Comment (benkio) [reply .scalafmt.conf]: https://github.com/apimorphism/iozhik/pull/6

File: project/plugins.sbt [hunk 3]
```diff
@@ -5,3 +5,4 @@ addSbtPlugin("com.github.sbt" % "sbt-pgp" % "2.1.2")
 addSbtPlugin("com.typesafe.sbt" % "sbt-git" % "1.0.1")
 addSbtPlugin("com.eed3si9n" % "sbt-buildinfo" % "0.10.0")
 addSbtPlugin("com.codecommit" % "sbt-github-actions" % "0.12.0")
+addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.4.0") // "2.4.0" is just sbt plugin version
```
Comment (johnspade) [inline project/plugins.sbt:8]: I think we should use 3.0.0-RC5

File: .scalafmt.conf (hunk 2 above)
Comment (johnspade) [reply .scalafmt.conf]: Thank you, it's merged. I modified the config a little: https://github.com/apimorphism/iozhik/blob/c68ec86436e0b45088e0854a5ad8b9841f0c10d2/iozhtest/src/main/resources/.scalafmt.conf, you can copy it and we'll merge this PR as well.

File: project/plugins.sbt (hunk 3 above)
Comment (benkio) [reply project/plugins.sbt:8]: That's the sbt plugin, not the actual scalafmt version:

https://github.com/scalameta/sbt-scalafmt/tags

**Pull request merged**